### PR TITLE
feat: platform pis stage first prototype

### PR DESCRIPTION
**Pull request by javfg**
This PR adds a first version of the PIS DAG for the Platform Pipeline, using Cloud Run jobs.

Things missing:

* Some PIS steps may fail because of credentials (like fetching google spreadsheets in `otar` step)

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -9,7 +9,7 @@ packages = [{include = "ot_orchestration", from = "src"}]
 
 [tool.poetry.dependencies]
 python = "^3.10, <3.11"
-apache-airflow = {extras = ["apache-beam", "async", "celery", "google", "kubernetes", "openlineage", "postgress", "sftp"], version = "^2.9.2"}
```
Comment (project-defiant) [inline pyproject.toml:12]: How strange , that poetry did allow for this

File: src/ot_orchestration/dags/platform_input_support.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    for step_name in get_steps():
+        name = clean_name(step_name)
+
+        with TaskGroup(group_id=name):
+            task_id = f"create_cloudrun_job_{step_name}"
+
+            @task(task_id=task_id)
+            def create_job(task_instance: TaskInstance | None = None):
+                """Create a Cloud Run job."""
+                job_name = f"{name}-{hash(task_instance.run_id)}"
```
Comment (project-defiant) [inline src/ot_orchestration/dags/platform_input_support.py]: Is hash required somehow for the logs?

File: src/ot_orchestration/dags/platform_input_support.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+                    job_name=job_name,
+                    job=create_job_instance(step_name),
+                    dag=dag,
+                )
+                c.execute(context=task_instance.get_template_context())
+
+            task_id = f"execute_cloudrun_job_{step_name}"
+
+            @task(task_id=task_id)
+            def execute_job(task_instance: TaskInstance | None = None):
```
Comment (project-defiant) [inline src/ot_orchestration/dags/platform_input_support.py:70]: The one thing I am missing is how is the `task_instance` passed to the `*_job` tasks. How airlfow knows that `task_instance` should be exactly TaskInstance.

File: src/ot_orchestration/dags/platform_input_support.py [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+                """Execute a Cloud Run job."""
+                job_name = f"{name}-{hash(task_instance.run_id)}"
```
Comment (project-defiant) [inline src/ot_orchestration/dags/platform_input_support.py]: I expect the job_name is different in create, execute and delete jobs?

**Review by project-defiant — APPROVED** [review]
Some minor comments, just for clarification.

File: src/ot_orchestration/dags/platform_input_support.py (hunk 2 above)
Comment (javfg) [reply src/ot_orchestration/dags/platform_input_support.py]: Yes. Also, this avoids cloud run job name collisions if two executions of the dag run in parallel.

File: src/ot_orchestration/dags/platform_input_support.py (hunk 3 above)
Comment (javfg) [reply src/ot_orchestration/dags/platform_input_support.py:70]: I use it to fetch the `run_id` for the currently running dag, as [the docs say](https://airflow.apache.org/docs/apache-airflow/stable/templates-ref.html#accessing-airflow-context-variables-from-taskflow-tasks).

Maybe there is an easier way to do this?

File: pyproject.toml (hunk 1 above)
Comment (javfg) [reply pyproject.toml:12]: Maybe it is failing silently on that extra install?

File: src/ot_orchestration/dags/platform_input_support.py (hunk 4 above)
Comment (javfg) [reply src/ot_orchestration/dags/platform_input_support.py]: Nope, `job_name` refers to the cloud run job, which has to be the same in the three tasks. The [`run_id`](https://airflow.apache.org/docs/apache-airflow/stable/_api/airflow/models/dagrun/index.html#airflow.models.dagrun.DagRun) is the same throughout the whole run of a DAG.

**Comment by javfg** [comment]
I'm organizing the code a bit more, I'll commit it ASAP.

**Comment by javfg** [comment]
@project-defiant just pushed those changes, you can take a look! :+1:

File: src/ot_orchestration/dags/platform_input_support.py (hunk 4 above)
Comment (project-defiant) [reply src/ot_orchestration/dags/platform_input_support.py]: make sense

**Review by project-defiant — APPROVED** [review]

**Comment by project-defiant** [comment]
@javfg you can merge when ready, nicely done!

**Pull request merged**